core/domain/draft_upgrade_services_test.py

```diff
@@ -120,6 +120,24 @@ def test_convert_to_latest_schema_version_implemented(self):
             msg='Current schema version is %d but DraftUpgradeUtil.%s is '
             'unimplemented.' % (state_schema_version, conversion_fn_name))
 
+    def test_convert_states_v30_dict_to_v31_dict(self):
```

Comment:
Have backend tests validating that this works correctly both for interactions that (a) are SetInput and (b) aren't SetInput.

Reply:
I've added tests addressing both cases.